### ISSUE TITLE
PHP7.1 nullable types not supported in doctrine:schema:update

### ISSUE THREAD

**Issue by fmonts** [doctrine/common#750]
I'm running PHP 7.1.0RC6 and I want to take advantage of the nullable types. 
(PHP 7.1 GA will be released on December 1)

This is a sample field in an entity:

```
/**
 * @ORM\Column(type="string", nullable=true)
 */
private $optionalField;


public function getOptionalField() : ?string
{
	return $this->optionalField;
}

public function setOptionalField(?string $optionalField)
{
	$this->optionalField = $optionalField;
}
```

Running doctrine:schema:update fails because of the following error:

> $ php bin/console doctrine:schema:update --dump-sql
> 
> 
>   [Symfony\Component\Debug\Exception\FatalThrowableError]
>   Parse error: syntax error, unexpected '?'

**Comment by stof** [comment]
@fmonts can you paste the full stack trace of the exception, so that we know where the error happened ?

**Comment by fmonts** [comment]
Sure:

> 
> Exception trace:
>  () at D:\project_path\src\AppBundle\Entity\Offer.php:169
>  Doctrine\Common\Persistence\Mapping\Driver\AnnotationDriver->getAllClassNames() at D:\project_path\vendor\doctrine\common\lib\Doctrine\Common\Persistence\Mapping\Driver\MappingDriverChain.php:128
>  Doctrine\Common\Persistence\Mapping\Driver\MappingDriverChain->getAllClassNames() at D:\project_path\vendor\doctrine\common\lib\Doctrine\Common\Persistence\Mapping\AbstractClassMetadataFactory.php:114
>  Doctrine\Common\Persistence\Mapping\AbstractClassMetadataFactory->getAllMetadata() at D:\project_path\vendor\doctrine\orm\lib\Doctrine\ORM\Tools\Console\Command\SchemaTool\AbstractCommand.php:59
>  Doctrine\ORM\Tools\Console\Command\SchemaTool\AbstractCommand->execute() at D:\project_path\vendor\doctrine\doctrine-bundle\Command\Proxy\UpdateSchemaDoctrineCommand.php:50
>  Doctrine\Bundle\DoctrineBundle\Command\Proxy\UpdateSchemaDoctrineCommand->execute() at D:\project_path\vendor\symfony\symfony\src\Symfony\Component\Console\Command\Command.php:254
>  Symfony\Component\Console\Command\Command->run() at D:\project_path\vendor\symfony\symfony\src\Symfony\Component\Console\Application.php:837
>  Symfony\Component\Console\Application->doRunCommand() at D:\project_path\vendor\symfony\symfony\src\Symfony\Component\Console\Application.php:187
>  Symfony\Component\Console\Application->doRun() at D:\project_path\vendor\symfony\symfony\src\Symfony\Bundle\FrameworkBundle\Console\Application.php:80
>  Symfony\Bundle\FrameworkBundle\Console\Application->doRun() at D:\project_path\vendor\symfony\symfony\src\Symfony\Component\Console\Application.php:118
>  Symfony\Component\Console\Application->run() at D:\project_path\bin\console:29
>

**Comment by Majkl578** [comment]
See doctrine/common#734, 7.1 support is implemented in master only, stable release should be released when 7.1.0 becomes GA. For now you can require `doctrine/common dev-master as 2.6.99` to try things out.

**Comment by fmonts** [comment]
ok I see, thank you
